public/pages/CreateMonitor/components/MonitorExpressions/expressions/GroupByExpression.js

```diff
@@ -69,32 +84,33 @@ class GroupByExpression extends Component {
         8 +
       60;
 
+    const isBucketLevelMonitor = monitorType === MONITOR_TYPE.BUCKET_LEVEL;
+
     if (
       (this.state.addButtonTouched || touched.groupBy) &&
       !values.groupBy.length &&
       values.monitor_type === MONITOR_TYPE.BUCKET_LEVEL
     )
       errors.groupBy = GROUP_BY_ERROR;
 
-    const { monitor_type: monitorType, groupBy } = values;
-
     let showAddButtonFlag = false;
-    if (MONITOR_TYPE.QUERY_LEVEL === monitorType && groupBy.length < 1) {
+    if (!isBucketLevelMonitor && groupBy.length < 1) {
```

Comment:
Isn't this a bug? Adding `!`?

Reply:
This is working as is. The monitor type is either Query level or Bucket level